File: mvn-advent-of-code/src/main/java/advent/twenty_fifteen/Day01NotQuiteLisp.java
package advent.twenty_fifteen;

import advent.common.DailyProblem;
import advent.utilities.FileUtilities;
import advent.utilities.Parsers;

public class Day01NotQuiteLisp implements DailyProblem<Integer, Integer> {

    private final int part1Answer;
    private final int part2Answer;

    public Day01NotQuiteLisp(String filename) {
        String line = FileUtilities.readLines(filename, Parsers::TO_STRING).get(0);

        int floor = 0;
        int position = 0;
        int firstBasementIndex = -1;
        for (char c : line.toCharArray()) {
            if (c == '(') {
                floor++;
            }
            if (c == ')') {
                floor--;
            }
            if (floor == -1 && firstBasementIndex == -1) {
                firstBasementIndex = position;
            }
            position++;
        }

        this.part1Answer = floor;
        this.part2Answer = firstBasementIndex+1;
    }

    @Override
    public Integer getPart1Answer() {
        return part1Answer;
    }

    @Override
    public Integer getPart2Answer() {
        return part2Answer;
    }
}

File: dotnet-advent-of-code/Aoc2015/Day01NotQuiteLisp.cs
using System;
using Common;

namespace TwentyFifteen
{
    public class Day01NotQuiteLisp : IDailyProblem<int, int>
    {
        public int Part1Answer { get; }
        public int Part2Answer { get; }

        public Day01NotQuiteLisp(string filename)
        {
            var line = InputFileReader.ReadLines(filename)[0];

            Part2Answer = -1;
            
            var floor = 0;
            var basementFound = false;
            for (int index = 0; index < line.Length; index++)
            {
                var c = line[index];
                floor += (c == '(') ? 1 : -1;

                if (floor == -1 && !basementFound)
                {
                    Part2Answer = index + 1;
                    basementFound = true;
                }
            }

            Part1Answer = floor;
        }
    }
}